concepts/box/about.md

```diff
@@ -1,3 +1,153 @@
 # About
 
-TODO
+To understand [`Box`][box] it is helpful to understand what the [stack and heap][stack and heap] are.
+
+The stack is an area of memory that works like a last-in-first-out (LIFO) buffer. It is very fast to push data onto the stack and it is fast to pop it off.
+The data can include variables or function addresses, but the data must be of a known fixed size at compilation time.
+
+If the data is of an unknown size at compilation time or has a size that may change, then it must be allocated on the heap. The allocation process must find
+space in the heap big enough to store the data, and then the address of where that data starts in memory gets pushed onto the stack. Retrieving data from the
+heap is slower because we need to go through the address on the stack and follow that to the location of tha data on the heap.
```

Comment:
This isn't wrong, but it's misleading. The major slowdown of the heap vs the stack is that allocating memory on the heap is very slow. Reads and mutation, as long as no reallocation is necessary, shouldn't be significantly slower than using the stack.

Reply:
Also just noticed a typo in that line: "tha" for "the".